.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+      - repo: local
+        hooks:
+              - id: clang-format
```

Comment:
It looks like you pulled this hook directly from cudf's pre-commit config. Regarding the discussion you and @jrhemstad are having on the PR about clang-format version mismatches, one way to address this would be to set up clang-format using [this hook](https://github.com/ssciwr/clang-format-wheel/) instead of using a system hook that just invokes whatever clang-format the user has installed. The upside is that this would allow us to specify a clang-format version, and a developer could just `pre-commit run clang-format --all-files` and be guaranteed to use a matching version. The downside is that a user invoking `clang-format` directly on the command line would use whatever version they had locally installed, so it could behave differently from invoking the pre-commit hook. IMO this is still preferable because it gives one guaranteed way for users to get the correct version of clang-format, but the mismatch could be confusing. What do you think?

Reply:
Depending on what you think of this suggestion we could get rid of the `run-clang-format.py` file since the pre-commit hook already includes the same functionality. You can invoke it on all files or a specified subset using command line arguments.

Reply:
My idea of keeping `run-clang-format.py` is to have one single command formatting all files. This is useful if a number of files are changed and users can just use this script to format them all without missing any files.

Reply:
voila!